File: main/209-20191015.rb
# @param {Integer} s
# @param {Integer[]} nums
# @return {Integer}
def min_sub_array_len(s, nums)
  size = nums.size
  min = size +1
  slow = 0
  fast = 0

  sum = 0
  for fast in 0...size do
    sum += nums[fast]

    while sum >= s
      min = [min, fast-slow+1].min
      sum -= nums[slow]
      slow += 1
    end

  end

  min == size + 1 ? 0 : min
end


s, nums = 7, [2,3,1,2,4,3]
# s, nums = 100, []
# s, nums = 11, [1,2,3,4,5]
puts min_sub_array_len(s, nums)

File: main/209-20191015.go
package main

func minSubArrayLen(s int, nums []int) int {
	sum, size := 0, len(nums)
	min := size + 1

	for slow, fast := 0, 0; fast < size; fast++ {
		sum += nums[fast]

		for sum >= s {
			if min > fast-slow+1 {
				min = fast - slow + 1
			}
			sum -= nums[slow]
			slow++
		}
	}

	if min == size+1 {
		return 0
	}
	return min
}

// func main() {
// 	// s := 7
// 	// nums := []int{2, 3, 1, 2, 4, 3}
// 	s, nums := 11, []int{1, 2, 3, 4, 5}
// 	fmt.Println(minSubArrayLen(s, nums))
// }
